src/core/infer_request.cc

```diff
@@ -199,7 +307,7 @@ InferenceRequest::Release(
   // and the callback may interact with upper level trace.
```

Comment:
Note that this layering will be broken now because setting trace_ = nullptr doesn't necessarily cause the trace release callback to be called. Not sure what the impact of this will be.

Reply:
I think this note is not necessary.
The layering will be the same as before. We only change the lifetime of trace, which controls the timing to write TraceStream into trace file. And we do not change any record timing. So, each record will stay the same sequence as before in the TraceStream.

Reply:
We will see once we start running some existing and new tests.